filecoin-proofs/src/api/post_adapter.rs

```diff
@@ -0,0 +1,626 @@
+    }
+
+    let mut dynamic = GeneratePoStDynamicSectorsCountOutput {
+        proofs: Vec::new(),
+        faults: Vec::new(),
+    };
+
+    for (i, x) in xs.into_iter().enumerate() {
+        match x {
+            Err(e) => {
+                return Err(e);
+            }
+            Ok(fixed) => {
+                dynamic.proofs.push(fixed.proof);
+                for fault in fixed.faults.iter() {
+                    dynamic.faults.push(((i as u64) * 2) + fault)
+                }
+            }
+        }
+    }
+
+    dynamic.faults = dynamic.faults.into_iter().take(orig_comm_rs_len).collect();
+
+    Ok(dynamic)
+}
+
+/// Maps inputs for a single dynamic sectors-count PoSt verify operation to a
+/// vector of PoSt verify inputs for fixed sectors-count.
+///
+pub fn verify_post_spread_input(
+    dynamic: VerifyPoStDynamicSectorsCountInput,
+) -> error::Result<Vec<VerifyPoStFixedSectorsCountInput>> {
+    let faults_len = dynamic.faults.len();
+    let commrs_len = dynamic.comm_rs.len();
+    let proofs_len = dynamic.proofs.len();
+    let replicas_c = (commrs_len as f64 / POST_SECTORS_COUNT as f64)
+        .ceil()
+        .round() as usize;
+    let faults_max = dynamic.faults.iter().max();
+
+    if faults_len > commrs_len {
+        return Err(format_err!(
+            "LEN(faults) must <= LEN(comm_rs): {:?}, {:?}",
+            faults_len,
+            commrs_len
+        ));
+    }
+
+    if proofs_len != replicas_c {
+        return Err(format_err!(
+            "LEN(proofs) must == CEIL(LEN(comm_rs)/2): {:?}, {:?}",
+            proofs_len,
+            replicas_c
+        ));
+    }
+
+    if faults_max
+        .map(|m| *m > commrs_len as u64 - 1)
+        .unwrap_or(false)
+    {
+        return Err(format_err!(
+            "MAX(faults) must <= LEN(comm_rs)-1: {:?}, {:?}",
+            faults_max,
+            commrs_len - 1
+        ));
+    }
+
+    let mut fixed: Vec<VerifyPoStFixedSectorsCountInput> = vec![];
+
+    let chunks = dynamic.comm_rs.chunks_exact(POST_SECTORS_COUNT);
+
+    let remainder = chunks.remainder();
+
+    // each POST_SECTORS_COUNT grouping of commitments maps to a single fixed
+    // sectors-count PoSt verification call
+    for (i, chunk) in chunks.enumerate() {
+        let mut comm_rs: [Commitment; POST_SECTORS_COUNT] = [[0; 32]; POST_SECTORS_COUNT];
+        for (i, comm_r) in chunk.iter().enumerate() {
+            comm_rs[i] = *comm_r;
+        }
+
+        fixed.push(VerifyPoStFixedSectorsCountInput {
+            sector_bytes: dynamic.sector_bytes,
+            comm_rs,
+            challenge_seed: dynamic.challenge_seed,
+            proof: dynamic.proofs[i],
+            faults: Vec::new(),
+        })
+    }
+
+    // If we receive a number of comm_rs which does not divide evenly by
+    // POST_SECTORS_COUNT, form a grouping of comm_rs which does divide
+    // evenly by POST_SECTORS_COUNT by duplicating the last comm_r in the
+    // group until LEN(comm_rs) == POST_SECTORS_COUNT.
```

Comment:
Yes, this is exactly the type of comment I had in mind that would help to understand the repetition format in `generate_`.